redis/connection.py

```diff
@@ -1278,7 +1281,7 @@ class ConnectionPool:
     """
 
     @classmethod
-    def from_url(cls, url, **kwargs):
+    def from_url(cls: Type[_CP], url: str, **kwargs) -> _CP:
```

Comment:
I think currently `cls` arg is used as str, it will be a breaking change

Reply:
@vladvildanov not sure I understood, where is it used as string? 😊

I copied the signature from here: https://github.com/redis/redis-py/blob/ceb12fe8b52f488bd1e0a42b9dc53161e77514d0/redis/asyncio/connection.py#L1005